src/FastField.tsx

```diff
@@ -78,20 +75,77 @@ export class FastField<
 
   handleChange = (e: React.ChangeEvent<any>) => {
     e.persist();
-    const { validateOnChange } = this.context.formik;
+    const {
+      validateOnChange,
+      validate,
+      values,
+      validationSchema,
+    } = this.context.formik;
     const { type, value, checked } = e.target;
     const val = /number|range/.test(type)
       ? parseFloat(value)
       : /checkbox/.test(type) ? checked : value;
-    if (this.props.validate && validateOnChange) {
-      const maybePromise = (this.props.validate as any)(value);
-      if (isPromise(maybePromise)) {
-        this.setState({ value: val });
-        (maybePromise as any).then(noop, (error: string) =>
-          this.setState({ error })
-        );
+    if (validateOnChange) {
+      // Field-level validation
+      if (this.props.validate) {
+        const maybePromise = (this.props.validate as any)(value);
+        if (isPromise(maybePromise)) {
+          this.setState({ value: val });
+          (maybePromise as any).then(
+            () => this.setState({ error: undefined }),
+            (error: string) => this.setState({ error })
+          );
+        } else {
+          this.setState({ value: val, error: maybePromise });
+        }
+      } else if (validate) {
+        // Top-level validate
+        const maybePromise = (validate as any)(value);
+        if (isPromise(maybePromise)) {
+          this.setState({ value: val });
+          (maybePromise as any).then(
+            () => this.setState({ error: undefined }),
+            (error: any) =>
+              this.setState({ error: getIn(error, this.props.name) })
+          );
+        } else {
+          this.setState({
+            value: val,
+            error: getIn(maybePromise, this.props.name),
```

Comment:
We could compare the error object returned by top-level validation / validationSchema to see if error object is different from context.errors. If it is, we could update top-level parent here too. With that added wrinkle, this could actually fully replace Field without breaking changes (I think)